server/cluster_worker_test.go

```diff
@@ -227,6 +226,129 @@ func (s *testClusterWorkerSuite) checkChangePeerRes(c *C, res *pdpb.RegionHeartb
 	}
 }
 
+func (s *testClusterWorkerSuite) askSplit(c *C, conn net.Conn, msgID uint64, r *metapb.Region) (uint64, []uint64) {
+	req := &pdpb.Request{
+		Header:  newRequestHeader(s.clusterID),
+		CmdType: pdpb.CommandType_AskSplit.Enum(),
+		AskSplit: &pdpb.AskSplitRequest{
+			Region: r,
+		},
+	}
+	sendRequest(c, conn, msgID, req)
+	_, resp := recvResponse(c, conn)
+	c.Assert(resp.GetCmdType(), Equals, pdpb.CommandType_AskSplit)
+	askResp := resp.GetAskSplit()
+	c.Assert(askResp, NotNil)
+	c.Assert(askResp.GetNewRegionId(), Not(Equals), 0)
+	c.Assert(askResp.GetNewPeerIds(), HasLen, len(r.Peers))
+	return askResp.GetNewRegionId(), askResp.GetNewPeerIds()
+}
+
+func updateRegionRange(r *metapb.Region, start, end []byte) {
+	r.StartKey = start
+	r.EndKey = end
+	r.RegionEpoch = &metapb.RegionEpoch{
+		ConfVer: proto.Uint64(r.GetRegionEpoch().GetConfVer()),
+		Version: proto.Uint64(r.GetRegionEpoch().GetVersion() + 1),
+	}
+}
+
+func splitRegion(c *C, old *metapb.Region, splitKey []byte, newRegionID uint64, newPeerIDs []uint64) *metapb.Region {
+	var peers []*metapb.Peer
+	c.Assert(len(old.Peers), Equals, len(newPeerIDs))
+	for i, peer := range old.Peers {
+		peers = append(peers, &metapb.Peer{
+			Id:      proto.Uint64(newPeerIDs[i]),
+			StoreId: proto.Uint64(peer.GetStoreId()),
+		})
+	}
+	newRegion := &metapb.Region{
+		Id:          proto.Uint64(newRegionID),
+		RegionEpoch: proto.Clone(old.RegionEpoch).(*metapb.RegionEpoch),
+		Peers:       peers,
+	}
+	updateRegionRange(newRegion, splitKey, old.EndKey)
+	updateRegionRange(old, old.StartKey, splitKey)
+	return newRegion
+}
+
+func (s *testClusterWorkerSuite) heartbeatRegion(c *C, conn net.Conn, msgID uint64, region *metapb.Region, leader *metapb.Peer) *pdpb.ChangePeer {
+	req := &pdpb.Request{
+		Header:  newRequestHeader(s.clusterID),
+		CmdType: pdpb.CommandType_RegionHeartbeat.Enum(),
+		RegionHeartbeat: &pdpb.RegionHeartbeatRequest{
+			Leader: leader,
+			Region: region,
+		},
+	}
+	sendRequest(c, conn, msgID, req)
+	_, resp := recvResponse(c, conn)
+	c.Assert(resp.GetCmdType(), Equals, pdpb.CommandType_RegionHeartbeat)
+	return resp.GetRegionHeartbeat().GetChangePeer()
+}
+
+func (s *testClusterWorkerSuite) TestSplit(c *C) {
```

Comment:
yep, but the probability is little for this case, I think we can do it later, can fire an issue first.